src/split-panel/bottom.tsx

```diff
@@ -45,7 +46,14 @@ export function SplitPanelContentBottom({
   const isMobile = useMobile();
 
   const headerRef = useRef<HTMLDivElement>(null);
-  useResizeObserver(headerRef, entry => reportHeaderHeight(entry.borderBoxHeight));
+  const closedPanelBlockSize = useRef<string>();
+
+  useResizeObserver(headerRef, entry => {
+    const { borderBoxHeight } = entry;
+    closedPanelBlockSize.current = `calc(${borderBoxHeight}px + ${tokens.borderPanelTopWidth})`;
```

Comment:
This line does not look like idiomatic React code. What is happening here?

Reply:
I am using `closedPanelBlockSize.current` as `blockSize` in line 86 in order to keep the animated transition. This gives the element the same height that it would already have without applying any value, _but_ a value needs to be explicitly set because you cannot transition from or to undefined; a value has to be defined for the transitioned property in both states.

What do you find most non-idiomatic here? Do you have any suggestion? Or does it look better knowing this information?

Reply:
For idiomatic code, it should be a state, not ref. Also, can we reuse `reportHeaderHeight` somehow which already sets the state?

Reply:
We can pass down `reportHeaderHeight` from the app layout, although it is only needed for the toolbar layout, so either we have to make it optional in the split panel context, or we pass it anyway in the three app layouts, whether it is actually used in the split panel or not.